api/column_family.cc

```diff
@@ -335,13 +335,13 @@ void set_column_family(http_context& ctx, routes& r) {
     cf::get_memtable_columns_count.set(r, [&ctx] (std::unique_ptr<request> req) {
         return map_reduce_cf(ctx, req->param["name"], 0, [](column_family& cf) {
             return cf.active_memtable().partition_count();
-        }, std::plus<int>());
+        }, std::plus<decltype(std::declval<column_family>().active_memtable().partition_count())>());
```

Comment:
Does `std::plus<>` work? C++17 made it a generic added.

I think it's also a good idea to change 0 to uint64_t(0). map_reduce_cf() deduces a lot from the initial value type.

Reply:
Yes. plus<> works: https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:29,endLineNumber:7,positionColumn:29,positionLineNumber:7,selectionStartColumn:29,selectionStartLineNumber:7,startColumn:29,startLineNumber:7),source:'%23include%3Cfunctional%3E%0A%23include%3Ciostream%3E%0A%0Aint+main()+%7B%0A++++constexpr+size_t+a+%3D+2147483648LL%3B%0A++++constexpr+size_t+b+%3D+1LL%3B%0A++++static_assert(std::plus%3C%3E()(a,+b)+%3D%3D+2147483649LL)%3B%0A++++std::cout+%3C%3C+std::plus%3C%3E()(a,+b)+%3C%3C+std::endl%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:clang1101,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'0',intel:'1',libraryCode:'0',trim:'1'),fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!((name:seastar,ver:'180')),options:'-std%3Dc%2B%2B20+-O3',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'x86-64+clang+11.0.1+(Editor+%231,+Compiler+%231)+C%2B%2B',t:'0')),k:50,l:'4',m:37.880633373934224,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compiler:1,editor:1,fontScale:14,fontUsePx:'0',wrap:'0'),l:'5',n:'0',o:'Output+of+x86-64+clang+11.0.1+(Compiler+%231)',t:'0')),header:(),l:'4',m:62.119366626065776,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4